File: subarray.java


// import java.util.*;
import java.io.*;
import java.util.*;
class Main{
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int [] arr = new int[scanner.nextInt()];
    int k = scanner.nextInt();
    for (int i = 0; i < arr.length;i++){
      arr[i] = scanner.nextInt();
    }
    System.out.println(slidingWindow(arr, k) - slidingWindow(arr, k - 1));


  } //main()
  public static  int slidingWindow(int[] nums, int k) {
        int count = 0, ans = 0;
        for (int begin = 0, end = 0; end < nums.length; end++) {
            if ((nums[end] & 1) == 1) {
                count++;
            }
            while (count > k) {
                if ((nums[begin++] & 1) == 1) {
                    count--;
                }
            }
            ans += end - begin + 1;
        }
        return ans;
    }
} // Main

File: subarray.py









def sub(start,end,arr,count,odd):
	if end > len(arr):
		return count
	temp = arr[start:end]
	c = 0
	for i in temp:
		if i % 2 == 1:
			c= c+ 1
	if c == odd :
		count =  count + 1
	return sub(start,end+1,arr,count,odd)


nums = [2,4,6]
k = 1
s = k
sum = 0
for i in range(len(nums)):
	x= sub(i,k,nums,0,s)
	sum = sum + x

print(sum)